docs/source/developers/guide/step_by_step/building.rst

```diff
@@ -155,6 +155,22 @@ Building other Arrow libraries
      `R developer docs <https://arrow.apache.org/docs/r/articles/developing.html>`_.
 
 
+     **Reinstalling R package and running 'make clean'**
+
+     When you will make changes to the Arrow C++ part of the code, also
+     called libarrow, you will need to:
+
+     #. reinstall libarrow
+     #. run ``make clean`` or ``git clean -fdx .``
+     #. reinstall the R package
+
+     ``make clean`` is a part of the ``r/Makefile`` added to help with some
+     common tasks from the command line (e.g. ``make test``, ``make doc``,
+     ``make clean``, etc.).
```

Comment:
I might rephrase this, as on a first read I misread it as meaning that `make clean` runs `make test` etc